be/src/olap/rowset/segment_creator.cpp

```diff
@@ -173,9 +190,10 @@ Status SegmentFlusher::_create_segment_writer(
     }
 
     writer = std::make_unique<segment_v2::VerticalSegmentWriter>(
-            file_writer.get(), segment_id, _context.tablet_schema, _context.tablet,
-            _context.data_dir, _context.max_rows_per_segment, writer_options, _context.mow_context);
-    RETURN_IF_ERROR(_seg_files.add(segment_id, std::move(file_writer)));
+            segment_file_writer.get(), segment_id, _context.tablet_schema, _context.tablet,
+            _context.data_dir, _context.max_rows_per_segment, writer_options, _context.mow_context,
+            std::move(inverted_file_v2_writer));
```

Comment:
You may need to do the same things for inverted_file_v2_writer as segment_file_writer. Just use inverted_file_v2_writer.get() here and add std::move(inverted_file_v2_writer) to _seg_files.